include/lsst/afw/typehandling/python.h

```diff
@@ -84,10 +88,121 @@ class StorableHelper : public Base {
     bool equals(Storable const& other) const noexcept override {
         PYBIND11_OVERLOAD_NAME(bool, Base, "__eq__", equals, other);
     }
+
+    bool isPersistable() const noexcept override {
+        PYBIND11_OVERLOAD(
+            bool, Base, isPersistable
+        );
+    }
+
+    std::string getPersistenceName() const override {
+        PYBIND11_OVERLOAD_NAME(
+            std::string, Base, "_getPersistenceName", getPersistenceName
+        );
+    }
+
+    std::string getPythonModule() const override {
+        PYBIND11_OVERLOAD_NAME(
+            std::string, Base, "_getPythonModule", getPythonModule
+        );
+    }
+
+    void write(table::io::OutputArchiveHandle& handle) const override;
 };
 
 std::string declareGenericMapRestrictions(std::string const& className, std::string const& keyName);
 
+/**
+ * StorableHelper persistence.
+ *
+ * We cannot directly override `StorableHelper::write` or `StorableHelperFactory::read` in python using
+ * PYBIND11_OVERLOAD* macros as the required argument c++ types (OutputArchiveHandle, InputArchive,
+ * CatalogVector) are not bound to python types.  Instead, we allow python subclasses of Storable to define
+ * `_write` and `_read` methods that return/accept a string serialization of the object.  The `_write` method
+ * should take no arguments (besides self) and return a string.  The `_read` method should be a static method
```

Comment:
I'd say "bytes" or "byte string" instead of "string" so nobody thinks we're talking about text-based persistence.